docs/semgrep-ci/manual-ci-setup.md

```diff
@@ -0,0 +1,414 @@
+3. If you are using Buildkite to store the configuration, save the file. Otherwise, commit the configuration file into the `/.buildkite` folder within the target repository.
+4. The Semgrep job starts automatically upon detecting the `pipelines.yml` commit in the previous step.
+5. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+:::note
+From Buildkite's main page, click **Pipelines > ➕ button** to perform these steps within Buildkite's UI.
+:::
+
+#### Buildkite code snippet
+
+```yaml
+- label: ":semgrep: Semgrep"
+  commands:
+    - export SEMGREP_RULES='p/default' 
+    - semgrep ci 
+  
+  plugins:
+    - docker#v3.7.0:
+        image: returntocorp/semgrep
+```
+
+### Running Semgrep in CI with other CI providers
+
+Do one of the following methods to run Semgrep in CI with other CI providers.
+
+#### Direct docker usage 
+
+Reference or add the [returntocorp/semgrep](https://hub.docker.com/r/returntocorp/semgrep) Docker image directly. The method to add the Docker image varies based on the CI provider.
+
+#### Install `semgrep`
+
+If you cannot use the Semgrep docker image, do the following steps.
+
+1. Run `pip install semgrep` inside the container or CI environment
+2. Run `semgrep ci --config auto`.
+
+Your customization options with other CI providers vary depending on working environment variables.
+
+## Refining the CI job
+
+The following sections describe methods to customize your CI job.
+
+### Passing the CI job 
+
+By default, Semgrep CI exits with exit code 1 if the scan returns any findings. This causes the job to fail.
+
+Semgrep provides a **fail open** option. This enables you to suppress findings or internal Semgrep errors that block your pipeline.  Use any of the following commands:
+
+<dl>
+	<dt><code>semgrep ci</code></dt>
+	<dd>The Semgrep CI job<strong>fails</strong> on blocking findings or on internal errors.</dd>
+	<dt><code>semgrep ci || [ $? != 1 ]</code></dt>
+	<dd>The Semgrep CI job <strong>fails</strong> on blocking findings, but <strong>passes</strong> on internal errors. </dd>
+	<dt><code>semgrep ci || true</code></dt>
+	<dd>The Semgrep CI job <strong>passes</strong> on blocking findings and on internal errors. </dd>
+</dl>
+
+Refer to [Semgrep exit codes](../cli-reference/#exit-codes) to understand various internal issues that cause Semgrep to fail.
+
+### Setting a scan schedule
+
+The following table is a summary of methods and resources to set up schedules for different CI providers.
+
+TODO
+
+### Customizing rules and rulesets
+
+#### Exploring Semgrep Registry for useful rulesets
+
+`semgrep ci` accepts a list of rules and rulesets to run on each scan. The list is delimited by a space (` `) if the variable is exported from a command or script block, such as in the BitBucket Pipeline example. The list can also be delimited by a newline if the `SEMGREP_RULES` variable is declared through YAML syntax.
```

Comment:
this whole paragraph is confusing to me. I don't know what you're trying to say., Maybe the examples will help, and then I can suggest alternative wording. or maybe this enough of a nudge for you to think of alternatives :-)

"such as in the BitBucket Pipeline example." -> link?